drivers/include/drv8825.hpp

```diff
@@ -47,5 +39,16 @@ class drv8825 : public hal::servo {
         //can lose resolution
         void set_position(double p_position);
         void driver_position(hal::degrees p_position);
+
+    private:
+        hal::output_pin& m_direction_pin;
+        hal::output_pin& m_step_pin;
+        hal::steady_clock& m_clock;
+        hal::time_duration m_step_half_period;
+        std::array<hal::output_pin*,3> m_mode_pins;
+
+        long m_partial_steps=0;// total 1/32 steps taken from pos 0 (so reverse directions steps subtract)
+        double m_partial_steps_to_deg=1;
```

Comment:
```suggestion
        // total 1/32 steps taken from pos 0 (so reverse directions steps subtract)
        long m_partial_steps = 0;
        float m_partial_steps_to_deg = 1;
```

Refrain from using doubles as most microcontrollers do NOT support doubles in hardware, resulting in performance losses by doing these operations in software. Float has the seam problem but works out better because most MCUs we plan to see will be 32-bit which fits in a single register. The double operations being in two separate registers adds to the complexity of the implementation and the machine instructions.